src/bindings/python/flux/util.py

```diff
@@ -63,10 +63,16 @@ def interruptible(func):
     """
 
     def func_wrapper(calling_obj, *args, **kwargs):
-        handler = signal.getsignal(signal.SIGINT)
-        signal.signal(signal.SIGINT, signal.SIG_DFL)
+        # python only allows `signal.signal` calls in the main thread
+        import threading
+
```

Comment:
Flux python developers will probably want the `import` here done at the top of the file (I've learned that one the hard way)

Reply:
> (I've learned that one the hard way)

Yeah, that's a combination of my and pylint's pedantry. Sorry for that.  I'm surprised to see that our CI tests pass with this import in the function.  I wonder if we turned that pylint check off, or worse, pylint isn't running.  Hmmm.

Reply:
Uh oh.  Looks like pylint warnings aren't fatal any longer 😨 I wouldn't worry about fixing that particular issue in this PR though.  It would be good to not introduce any new warnings though.

Pylint did in fact complain about this import: https://travis-ci.com/github/flux-framework/flux-core/jobs/399160607#L258.  Can it be moved up to the top of the file?

Pylint is also complaining about the accessing of `threading._MainThread`.  We only support python 3.6+, so you can leverage the `threading.main_thread()` method, which was added in 3.4:  https://docs.python.org/3.4/library/threading.html#threading.main_thread